### ISSUE TITLE
Install on windows

### ISSUE THREAD

**Issue by dfalster** [traitecoevo/plant#145]

**Comment by richfitz** [comment]
It looks like the most recent version of rtools (still under development) might work http://cran.r-project.org/bin/windows/Rtools/

It includes gcc 4.9.2 which supports C++11.  It *was* going to be introduced for 3.2.0 but there were too many test failures so they backed out.  That means that it might be supported for 3.2.1 (pre-releases are coming out in June).

I've confirmed that it does *not* work with the default toolchain on 3.2.0 (via win builder)

```
g++  -std=c++0x -I"D:/RCompile/recent/R-3.2.0/include"         -I../inst/include/   -I"d:/RCompile/CRANpkg/lib/3.2/Rcpp/include" -I"d:/RCompile/CRANpkg/lib/3.2/BH/include" -I"d:/RCompile/r-compiling/local/local320/include"     -O2 -Wall  -mtune=core2 -c RcppExports.cpp -o RcppExports.o
In file included from ../inst/include/tree.h:9:0,
                 from RcppExports.cpp:4:
../inst/include/tree/interpolator.h:12:7: warning: 'tree::interpolator::Interpolator' has a field 'tree::interpolator::Interpolator::tk_spline' whose type uses the anonymous namespace [enabled by default]
g++  -std=c++0x -I"D:/RCompile/recent/R-3.2.0/include"         -I../inst/include/   -I"d:/RCompile/CRANpkg/lib/3.2/Rcpp/include" -I"d:/RCompile/CRANpkg/lib/3.2/BH/include" -I"d:/RCompile/r-compiling/local/local320/include"     -O2 -Wall  -mtune=core2 -c RcppR6.cpp -o RcppR6.o
In file included from ../inst/include/tree.h:9:0,
                 from RcppR6.cpp:2:
../inst/include/tree/interpolator.h:12:7: warning: 'tree::interpolator::Interpolator' has a field 'tree::interpolator::Interpolator::tk_spline' whose type uses the anonymous namespace [enabled by default]
In file included from ../inst/include/tree.h:31:0,
                 from RcppR6.cpp:2:
../inst/include/tree/patch.h: In lambda function:
../inst/include/tree/patch.h:142:16:   instantiated from 'tree::Patch<T>::compute_light_environment() [with T = tree::Plant<tree::FFW16_Strategy>]::<lambda(double)>'
../inst/include/tree/patch.h:142:66:   instantiated from 'void tree::Patch<T>::compute_light_environment() [with T = tree::Plant<tree::FFW16_Strategy>]'
../inst/include/tree/patch.h:67:39:   instantiated from 'void tree::Patch<T>::r_compute_light_environment() [with T = tree::Plant<tree::FFW16_Strategy>]'
RcppR6.cpp:935:37:   instantiated from here
../inst/include/tree/patch.h:142:64: error: 'canopy_openness' was not declared in this scope
../inst/include/tree/patch.h: In lambda function:
../inst/include/tree/patch.h:150:16:   instantiated from 'tree::Patch<T>::rescale_light_environment() [with T = tree::Plant<tree::FFW16_Strategy>]::<lambda(double)>'
../inst/include/tree/patch.h:150:66:   instantiated from 'void tree::Patch<T>::rescale_light_environment() [with T = tree::Plant<tree::FFW16_Strategy>]'
../inst/include/tree/patch.h:223:5:   instantiated from 'tree::ode::const_iterator tree::Patch<T>::set_ode_state(tree::ode::const_iterator, double) [with T = tree::Plant<tree::FFW16_Strategy>, tree::ode::const_iterator = __gnu_cxx::__normal_iterator<const double*, std::vector<double> >]'
../inst/include/tree/ode_interface.h:113:3:   instantiated from 'typename std::enable_if<tree::ode::needs_time<T>::value, void>::type tree::ode::r_set_ode_state(T&, const state_type&, double) [with T = tree::Patch<tree::Plant<tree::FFW16_Strategy> >, typename std::enable_if<tree::ode::needs_time<T>::value, void>::type = void, tree::ode::state_type = std::vector<double>]'
RcppR6.cpp:947:49:   instantiated from here
../inst/include/tree/patch.h:150:64: error: 'canopy_openness' was not declared in this scope
../inst/include/tree/patch.h: In lambda function:
../inst/include/tree/patch.h:142:66: warning: control reaches end of non-void function [-Wreturn-type]
../inst/include/tree/patch.h: In lambda function:
../inst/include/tree/patch.h:150:66: warning: control reaches end of non-void function [-Wreturn-type]
make: *** [RcppR6.o] Error 1
```

**Comment by dfalster** [comment]
Apparently the latest version of R (3.3.0) includes a suitable compiler so one can compile locally. Installation via drat also works well. Running the tests shows a few warnings (that could be present in general but have previously been hidden in testthat) and one failure with at test-plant-runner.R#76 which could also be a testthat failure.

**Comment by RemkoDuursma** [comment]
I tried this as well, and can confirm that with R 3.3.0 the package can be installed (and it works).

**Comment by richfitz** [comment]
I suggest we increase the required R version to 3.3.0, and document installation of Rtools (which is really easy).  Then put the binary in the `traitecoevo` drat (build it on win-builder, download the binary and do `drat::insertPackage(<name of .zip>)`)

In the meantime, the tests probably need tidying up.  I've cleaned several thousand tests over to new testthat style (in addition to the current set of breaking API changes I suspect more are in the pipeline :fire::fire::fire::fire::fire::fire::fire:).

**Comment by richfitz** [comment]
Also would be worth setting up appveyor.  That should be fairly straightforward so long as it uses the current rtools (I think it does)

**Comment by dfalster** [comment]
Ok, [first attempt failed](https://ci.appveyor.com/api/buildjobs/rhfgwcgi84qe7grm/log) because need to install loggr

```
Package required but not available: 'loggr'

Packages suggested but not available: 'RcppR6' 'plant.ml'
```

**Comment by dfalster** [comment]
Ok, next attempt "failed" because  appveyor definitely wants the dependencies `plant.ml`:

```
* checking package dependencies ... ERROR 
Package suggested but not available: 'plant.ml'
The suggested packages are required for a complete check.
```

But we can ignore those by

```
The suggested packages are required for a complete check.
Checking can be attempted without them by setting the environment
variable _R_CHECK_FORCE_SUGGESTS_ to a false value.
```

**Comment by dfalster** [comment]
Ok, we've made it through to havig appveyor actually run the tests (yay), but encountering some failures:

```
* checking tests ... ERROR
Running the tests in 'tests/testthat.R' failed.
Last 13 lines of output:
  OK: 1303 SKIPPED: 1 FAILED: 9
  1. Failure: Ported from tree1 (@test-cohort.R#60) 
  2. Failure: ODE interface (@test-cohort.R#120) 
  3. Failure: Gradients agree (@test-gradient.R#34) 
  4. Failure: Sensible behaviour on integration failure (@test-plant-runner.R#172) 
  5. Failure: Sensible behaviour on integration failure (@test-plant-runner.R#177) 
  6. Failure: Reference comparison (@test-plant.R#81) 
  7. Failure: Vectorised interface to integration works (@test-qk.R#59) 
  8. Failure: Vectorised interface to integration works (@test-qk.R#67) 
  9. Failure: Vectorised interface to integration works (@test-qk.R#71) 
  
  Error: testthat unit tests failed
  Execution halted
* DONE
Status: 1 ERROR, 3 WARNINGs, 3 NOTEs
+ DumpLogsByExtension fail
+ [[ -z fail ]]
+ extension=fail
+ shift
++ find . -maxdepth 1 -name '*.Rcheck' -type d
+ package=./plant.Rcheck
+ [[ 1 -ne 1 ]]
++ find ./plant.Rcheck -type f -name '*fail'
+ for name in '$(find "${package}" -type f -name "*${extension}")'
+ echo '>>> Filename: ./plant.Rcheck/tests/testthat.Rout.fail <<<'
>>> Filename: ./plant.Rcheck/tests/testthat.Rout.fail <<<
+ cat ./plant.Rcheck/tests/testthat.Rout.fail
 
R version 3.3.0 Patched (2016-06-02 r70706) -- "Supposedly Educational"
Copyright (C) 2016 The R Foundation for Statistical Computing
Platform: i386-w64-mingw32/i386 (32-bit)
 
R is free software and comes with ABSOLUTELY NO WARRANTY.
You are welcome to redistribute it under certain conditions.
Type 'license()' or 'licence()' for distribution details.
 
R is a collaborative project with many contributors.
Type 'contributors()' for more information and
'citation()' on how to cite R or R packages in publications.
 
Type 'demo()' for some demos, 'help()' for on-line help, or
'help.start()' for an HTML browser interface to help.
Type 'q()' to quit R.
 
> library(testthat)
> library(plant)
> 
> test_check("plant")
1. Failure: Ported from tree1 (@test-cohort.R#60) ------------------------------
`dgdh` not identical to `dgdh_forward`.
Objects equal but not identical
 
 
2. Failure: ODE interface (@test-cohort.R#120) ---------------------------------
cohort$plant$internals[v] not identical to plant$internals[v].
Objects equal but not identical
 
 
3. Failure: Gradients agree (@test-gradient.R#34) ------------------------------
test_gradient_richardson(f, x, d, r) not identical to numDeriv::grad(f, x, method.args = method_args).
Objects equal but not identical
 
 
4. Failure: Sensible behaviour on integration failure (@test-plant-runner.R#172) 
`warnings` does not match "integration failed with error".
Actual value: "Time exceeded time_max, 50 larger sizes dropped"
 
 
5. Failure: Sensible behaviour on integration failure (@test-plant-runner.R#177) 
nrow(res$trajectory) not equal to 1.
1/1 mismatches
[1] 111 - 1 == 110
 
 
6. Failure: Reference comparison (@test-plant.R#81) ----------------------------
vars_pl[rate_names] not identical to vars_pp[rate_names].
Objects equal but not identical
 
 
7. Failure: Vectorised interface to integration works (@test-qk.R#59) ----------
int_21$integrate_vector(f(x), a, b) not identical to int_21$integrate(f, a, b).
Objects equal but not identical 
 
 
8. Failure: Vectorised interface to integration works (@test-qk.R#67) ----------
int_41$integrate_vector(f(x), a, b) not identical to int_41$integrate(f, a, b).
Objects equal but not identical
 
 
9. Failure: Vectorised interface to integration works (@test-qk.R#71) ----------
int_51$integrate_vector(f(x), a, b) not identical to int_51$integrate(f, a, b).
Objects equal but not identical
 
 
testthat results ================================================================
OK: 1303 SKIPPED: 1 FAILED: 9
1. Failure: Ported from tree1 (@test-cohort.R#60) 
2. Failure: ODE interface (@test-cohort.R#120) 
3. Failure: Gradients agree (@test-gradient.R#34) 
4. Failure: Sensible behaviour on integration failure (@test-plant-runner.R#172) 
5. Failure: Sensible behaviour on integration failure (@test-plant-runner.R#177) 
6. Failure: Reference comparison (@test-plant.R#81) 
7. Failure: Vectorised interface to integration works (@test-qk.R#59) 
8. Failure: Vectorised interface to integration works (@test-qk.R#67) 
9. Failure: Vectorised interface to integration works (@test-qk.R#71) 
```

Here's the [direct link to the appveyor build](https://ci.appveyor.com/project/dfalster/plant/build/1.0.4). The same commit [is passing on Travis-CI](https://travis-ci.org/traitecoevo/plant/builds/135498640) and on my local machine. Without having a windows machine this will be hard to followup. @richfitz could you possibly take a look?

**Comment by dfalster** [comment]
Now works on windows. I updated the description & readme, bumped dependency and added appveyor (now passing, #184). Have removed drat from installation instructions as I'm not sure I'll support that long term
